File: 第11章/11-2/Project1/Rectangle.cpp
#include "Rectangle.h"

Rectangle::Rectangle()
{
	width = 1;
	height = 1;
}

Rectangle::Rectangle(double width, double height)
{
	this->width = width;
	this->height = height;
}

void Rectangle::setWidth(double width)
{
	this->width = width;
}

void Rectangle::setHeight(double height)
{
	this->height = height;
}

double Rectangle::getArea()
{
	return height * width;
}

File: python/GeometricObject_Inheritance/PythonApplication1/Rectangle.py
from GeometricObject import GeometricObject

class Rectangle(GeometricObject):

    def __init__(self, width , height , color , filled):
        super().__init__(color, filled)
        self.height = height
        self.width = width

    def setHeight(self, height):
        self.height = height

    def getHeight(self):
        return self.height

    def setWidth(self, width):
        self.width = width

    def getWidth(self):
        return self.width

    def getArea(self):
        return self.height * self.width

    def getPerimeter(self):
        return (self.height + self.width) * 2
